README.md

```diff
@@ -40,6 +40,12 @@ if (verified) {
 }
 ```
 
+Use the library for 3ID authenticate:
+
+```js
+await authenticate.ethereum(message, '0x123abc...', ethereumProvider)
```

Comment:
This api seems wrong. We should be able to have a generic authenticate function in the same way as `createLink` and `validateLink` are generic. The `createLink` function uses a type detector, we should be able to use that. This also aligns us well to the update to CAIP-10 where we can just detect the auth function from the accountId.